diff -up tbb2017_20161128oss/build/linux.inc.s390x tbb2017_20161128oss/build/linux.inc
--- tbb2017_20161128oss/build/linux.inc.s390x	2016-11-29 15:24:54.000000000 +0100
+++ tbb2017_20161128oss/build/linux.inc	2017-01-02 10:28:44.000000000 +0100
@@ -99,7 +99,10 @@ SONAME_SUFFIX =$(shell grep TBB_COMPATIB
 ifeq ($(arch),ia64)
         def_prefix = lin64ipf
 endif
-ifeq ($(arch),sparc s390x)
+ifeq ($(arch),sparc)
+        def_prefix = lin64
+endif
+ifeq ($(arch),s390x)
         def_prefix = lin64
 endif
 ifeq ($(arch),armv7)
