diff -up tbb2017_20160916oss/build/linux.inc.s390x tbb2017_20160916oss/build/linux.inc
--- tbb2017_20160916oss/build/linux.inc.s390x	2016-10-07 13:54:30.422343446 +0200
+++ tbb2017_20160916oss/build/linux.inc	2016-10-07 13:54:53.002343446 +0200
@@ -102,6 +102,9 @@ endif
 ifeq ($(arch),sparc)
         def_prefix = lin64
 endif
+ifeq ($(arch),s390x)
+        def_prefix = lin64
+endif
 ifeq ($(arch),armv7)
         def_prefix = lin32
 endif
